Sources/InternxtSwiftCore/Services/HttpAPI/BackupAPI.swift

```diff
@@ -38,6 +38,16 @@ public struct BackupAPI {
         return try await apiClient.fetch(type: DeviceAsFolder.self, endpoint, debugResponse: debug)
     }
 
+    public func editDeviceName(deviceId: Int, deviceName: String, debug: Bool = false) async throws -> DeviceAsFolder {
+        let endpoint = Endpoint(
+            path: "\(self.baseUrl)/backup/deviceAsFolder/\(deviceId)",
+            method: .PATCH,
+            body: CreateDevicePayload(deviceName: deviceName).toJson()
```

Comment:
create an `EditDevicePayload` as we are probably not going to allow to edit the same properties we use to create a device, and we may face problems in the future